### PR TITLE
R2dbc rc2

### LINKED ISSUE #152: Upgrade to R2DBC 0.8.0.RC1
R2DBC 0.8.0.RC1 is [scheduled for the end of September](https://github.com/r2dbc/r2dbc-spi/milestone/6). It makes sense to take a look at most recent SPI snapshots to catch up with the SPI changes.

### PR DESCRIPTION
**Pull request by oshai**
fixes #152 
Upgraded directly to RC2 - skipped RC1

### CODE REVIEW

File: r2dbc-mysql/src/main/java/JasyncClientConnection.kt [hunk 1]
```diff
@@ -21,6 +47,15 @@ class JasyncClientConnection(private val jasyncConnection: JasyncConnection) : C
         return executeVoid("COMMIT")
     }
 
+    override fun isAutoCommit(): Boolean {
+        return autoCommit
+    }
+
+    override fun setAutoCommit(autoCommit: Boolean): Publisher<Void> {
+        this.autoCommit = autoCommit
```
Comment (mp911de) [inline r2dbc-mysql/src/main/java/JasyncClientConnection.kt]: Setting auto-commit state should be deferred (i.e. on completion or on subscribe of the `SET AUTOCOMMIT` statement)

File: r2dbc-mysql/src/main/java/JasyncClientConnection.kt [hunk 2]
```diff
@@ -41,9 +76,14 @@ class JasyncClientConnection(private val jasyncConnection: JasyncConnection) : C
     }
 
     override fun setTransactionIsolationLevel(isolationLevel: IsolationLevel): Publisher<Void> {
+        this.isolationLevel = isolationLevel
```
Comment (mp911de) [inline r2dbc-mysql/src/main/java/JasyncClientConnection.kt]: Setting should be deferred. See auto-commit comment.

File: r2dbc-mysql/src/main/java/JasyncClientConnection.kt (hunk 1 above)
Comment (oshai) [reply r2dbc-mysql/src/main/java/JasyncClientConnection.kt]: I changed it in latest commit, thanks!

File: r2dbc-mysql/src/main/java/JasyncClientConnection.kt (hunk 2 above)
Comment (oshai) [reply r2dbc-mysql/src/main/java/JasyncClientConnection.kt]: I couldn't find this information on the server status so kept it as is.

**Pull request merged**